spirv_msl.cpp

```diff
@@ -393,6 +428,15 @@ uint32_t CompilerMSL::add_interface_block(StorageClass storage)
 	MemberSorter memberSorter(ib_type, meta[ib_type_id], sort_aspect);
 	memberSorter.sort();
 
+	// Sort input or output variables alphabetical
+	auto &execution = get_entry_point();
+	if ((execution.model == ExecutionModelFragment && storage == StorageClassInput) ||
+	    (execution.model == ExecutionModelVertex && storage == StorageClassOutput))
+	{
+		MemberSorter memberSorter(ib_type, meta[ib_type.self], MemberSorter::Alphabetical);
+		memberSorter.sort();
+	}
```

Comment:
Hmm, thats exactly why i sort alphabetically. I don’t know how exactly members are sorted now. But in `CompilerGLSL::emit_struct` an index i is used to set the position, and since members in vertex and fragment shaders are not in the same order, this won’t work. Try it with [painter-text.frag.glsl](https://github.com/Kode/krafix_tests/blob/a5aefcd66fb1ddc575cfb5a38e24587319ff49d4/in/painter-text.frag.glsl) and [painter-text.vert.glsl](https://github.com/Kode/krafix_tests/blob/a5aefcd66fb1ddc575cfb5a38e24587319ff49d4/in/painter-text.vert.glsl). Without sorting alphabetically i get for vertex:
```
struct painter_text_vert_main_out
{
    float2 texCoord [[user(locn0)]];
    float4 fragmentColor [[user(locn1)]];
    float4 gl_Position [[position]];
};
```

and fragment:
struct painter_text_frag_main_in
```
{
    float4 fragmentColor [[user(locn0)]];
    float2 texCoord [[user(locn1)]];
}
```

The fragment input does not match the corresponding vertex shader output. Perhaps i should better look how members are ordered until now.